finite_difference/src/Makefile

```diff
@@ -81,19 +92,31 @@ SOURCES = $(BASE_SOURCES:%=$(ROOT)/%)
 MODULES := $(BASE_SOURCES:%.f90=%.o)
 MODULES := $(MODULES:%.F90=%.o)
 
-.PHONY: install clean all
+# Location of the FortCL library (git sub-module)
+FORTCL_DIR = $(ROOT)/../../external/FortCL/src
+
+.PHONY: install clean all distclean
 
 all: ${API_LIB}
 
-${API_LIB}: ${MODULES}
-	${AR} ${ARFLAGS} ${API_LIB} ${MODULES}
+# Create the archive. Note that we include the FortCL objects in this
+# archive but take care to exclude clfortran.o (as that only has
+# interface definitions, not implementation).
+${API_LIB}: fortcl ${MODULES}
+	${AR} ${ARFLAGS} ${API_LIB} ${FORTCL_DIR}/*.o ${MODULES}
+	${AR} d ${API_LIB} clfortran.o
 
+fortcl:
+	${MAKE} -C ${FORTCL_DIR} ${FORTCL_TARGET}
```

Comment:
While ``make F90=gfortran`` works as expected, just doing ``make`` does not (it somehow ends up with F90='c'.

Adding ``F90=$(F90)`` to the make call  seems to work fine.